### PR TITLE
chore: add vuln report generate event

### PR DESCRIPTION
**Pull request by hectorj2f**
part of this PR: https://github.com/chainguard-dev/mono/pull/15340

### CODE REVIEW

File: events/generator/generator.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	// For example, "ColumbusOHUS" or "Minato City13JP".
+	Location string `json:"location"`
+
+	// UserAgent holds the user-agent of the client who pushed.
+	UserAgent string `json:"user_agent"`
+
+	Error *Error `json:"error,omitempty"`
+
+	// RepoManifestDigestID identifies the UIDP of the repo_manifest_digest being pushed
+	RepoManifestDigestID string `json:"repo_manifest_digest_id"`
```
Comment (wlynch) [inline events/generator/generator.go:41]: These are the output values? Should these be separated into a separate type?

somewhat related - you may find https://github.com/googleapis/googleapis/blob/2610c7a8f3c362f1ef54cb0bb636ded47422eec8/google/longrunning/operations.proto#L128-L162, https://google.aip.dev/151 interesting - lots of similarities here.

File: events/generator/generator.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	"chainguard.dev/sdk/civil"
+)
+
+const GenerateVulnReportEventType = "dev.chainguard.vulnreport.generate.v1"
+
+type Error struct {
+	Status  int    `json:"status"`
+	Code    string `json:"code"`
+	Message string `json:"message"`
+}
```
Comment (wlynch) [inline events/generator/generator.go:13]: Should we use https://github.com/googleapis/googleapis/blob/2610c7a8f3c362f1ef54cb0bb636ded47422eec8/google/rpc/status.proto#L35 / https://pkg.go.dev/google.golang.org/grpc/status#Status instead? 🤔

File: events/generator/generator.go (hunk 1 above)
Comment (hectorj2f) [reply events/generator/generator.go:41]: I am not going to use it so we can omit it.

File: events/generator/generator.go (hunk 2 above)
Comment (hectorj2f) [reply events/generator/generator.go]: I changed the code now to use this type.

File: events/generator/generator.go (hunk 1 above)
Comment (hectorj2f) [reply events/generator/generator.go]: I decided to simplify the type itself.

**Review by wlynch — APPROVED** [review]

**Pull request merged**